src/PostHog.AI/PostHogOpenAIHandler.cs

```diff
@@ -0,0 +1,744 @@
+        )
+        {
+            var read = await _innerStream
+                .ReadAsync(buffer, cancellationToken)
+                .ConfigureAwait(false);
+            if (read > 0)
+            {
+                if (MemoryMarshal.TryGetArray(buffer, out ArraySegment<byte> segment))
+                {
+                    ProcessChunk(segment.Array!, segment.Offset, read);
+                }
+                else
+                {
+                    // Fallback for non-array-backed Memory<byte>
+                    var temp = buffer.Slice(0, read).ToArray();
+                    ProcessChunk(temp, 0, read);
+                }
+            }
+
+            return read;
+        }
+
+        private void ProcessChunk(byte[] buffer, int offset, int count)
+        {
+#pragma warning disable CA1031 // Do not catch general exception types
+            try
+            {
+                var chunk = Encoding.UTF8.GetString(buffer, offset, count);
+
+                // Append chunk to line buffer
+                _lineBuffer.Append(chunk);
+
+                // Process complete SSE messages (terminated by \n\n or \r\n\r\n)
+                while (true)
+                {
+                    var bufferText = _lineBuffer.ToString();
+                    if (string.IsNullOrEmpty(bufferText))
+                        break;
+
+                    // Look for complete SSE message (terminated by \n\n or \r\n\r\n)
+                    var messageEnd = bufferText.IndexOf("\n\n", StringComparison.Ordinal);
+                    var lineEndLength = 2;
+
+                    if (messageEnd == -1)
+                    {
+                        messageEnd = bufferText.IndexOf("\r\n\r\n", StringComparison.Ordinal);
+                        if (messageEnd == -1)
+                        {
+                            // No complete message found, keep remaining buffer
+                            break;
+                        }
+                        lineEndLength = 4;
+                    }
+
+                    // Extract complete message (including the line ending)
+                    var messageLength = messageEnd + lineEndLength;
+                    var message = bufferText.Substring(0, messageLength);
+
+                    // Remove processed message from buffer
+                    _lineBuffer.Remove(0, messageLength);
+
+                    // Process the complete SSE message
+                    ProcessSSEMessage(message);
+                }
+            }
+            catch (Exception)
+            {
+                // Ignore other errors during chunk processing to avoid crashing the stream.
+                // The stream should still flow to the client.
+            }
+#pragma warning restore CA1031
+        }
+
+        private void ProcessSSEMessage(string message)
+        {
+            // Split message into lines
+            var lines = message.Split(LineSeparators, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
This allocates a new string array for each SSE message chunk. For streaming responses with 50-100+ chunks, this adds up. Since you're targeting net6.0+, you could use span-based enumeration:

```csharp  
foreach (var line in message.AsSpan().EnumerateLines())
```